config/fi_check_package.m4

```diff
@@ -130,7 +130,7 @@ AC_DEFUN([_FI_CHECK_PACKAGE_LIB], [
                           unset fi_Lib])])])])
 
     AS_IF([test "$fi_check_package_lib_happy" = "yes"],
-          [$1_LIBS="-l$2 $4"
+          [$1_LIBS="-l$2 $4 $$1_LIBS"
```

Comment:
I thought the prefix intended to be used was the provider name. But maybe that's what is wrong here.